### PR TITLE
[Do Not Merge] log in if tgt is specified to make running as a service easier.

### PR DESCRIPTION
**Pull request by jmontleon**

### CODE REVIEW

**Comment by jmontleon** [comment]
Without this running as a service currently requires doing:
```
sudo su -s /bin/bash - ansibleservicebroker
oc login
```
This removes the need for this manual intervention. Simply place credentials in /etc/ansible-service-broker/broker-config.yaml. If there is no openshift/target key nothing will be attempted, do I wouldn't expect it to interfere or cause an issue when running inside a container using credentials pulled in from the secret.

**Comment by jwmatthews** [comment]
Would like to discuss some alternatives before we consider merging this.

File: extras/ansible-service-broker.spec [hunk 1]
```diff
@@ -53,6 +53,8 @@ BuildRequires: %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
 
 Requires(pre): shadow-utils
 Requires: %{name}-selinux
+#Listing the file because the upstream and downstream package names differ
```
Comment (jmrodri) [inline extras/ansible-service-broker.spec]: add space after # to be consistent with other comments.

File: extras/ansible-service-broker.spec [hunk 2]
```diff
@@ -53,6 +53,8 @@ BuildRequires: %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
...
+Requires: /usr/bin/oc
```
Comment (jmrodri) [inline extras/ansible-service-broker.spec]: +1

File: extras/ansible-service-broker.spec [hunk 3]
```diff
@@ -65,6 +67,7 @@ BuildRequires: btrfs-progs-devel
 %package container-scripts
 Summary: scripts required for running ansible-service-broker in a container
 BuildArch: noarch
+Requires: %{name}
```
Comment (jmrodri) [inline extras/ansible-service-broker.spec]: +1

File: build/ansible-service-broker [hunk 4]
```diff
@@ -1,5 +1,36 @@
 #!/usr/bin/env bash
 
-BROKER_CONFIG=/etc/ansible-service-broker/broker-config.yaml
+export BROKER_CONFIG=/etc/ansible-service-broker/broker-config.yaml
+
+python - <<END
+import os
+import yaml
+from subprocess import call
```
Comment (jmrodri) [inline build/ansible-service-broker]: Will `yaml` need to be a requirement in the rpm or is it usually installed with python?

File: build/ansible-service-broker [hunk 5]
```diff
@@ -1,5 +1,36 @@
 #!/usr/bin/env bash
 
-BROKER_CONFIG=/etc/ansible-service-broker/broker-config.yaml
+export BROKER_CONFIG=/etc/ansible-service-broker/broker-config.yaml
+
+python - <<END
```
Comment (jmrodri) [inline build/ansible-service-broker]: that's really clever, didn't know you could do this.

File: build/ansible-service-broker [hunk 6]
```diff
@@ -1,5 +1,36 @@
...
+    user=config['openshift']['user']
+    passwd=config['openshift']['pass']
+  except Exception as e:
+    print("Could not read openshift configuration options. Error was: "+format(e)+"\n")
+    exit(1)
+
+  try:
+    call(["oc", "login", "--server=https://"+target, "-u="+user, "-p="+passwd, "--insecure-skip-tls-verify=true" ])
+  except Exception as e:
+    print("Did not successfully run oc login. Error was: "+format(e)+"\n")
```
Comment (jmrodri) [inline build/ansible-service-broker]: I'd rephrase the error message.
```python
print("Unable to run oc login. Error was: "+format(e)+"\n")
```

I fear that the `not` would missed. I had to read the sentence twice to notice it. I thought it said `Did successfully run oc login`

**Comment by eriknelson** [comment]
Closing, support for OPENSHIFT_{TARGET,USER,PASS} has been removed from the broker.
